File: problems/2610_convert_an_array_into_a_2d_array_with_conditions/solution.py
from typing import List


class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        from collections import Counter
        cnt = Counter(nums)
        res = []
        while cnt:
            row = list(cnt.keys())
            res.append(row)
            for k in row:
                cnt[k] -= 1
                if cnt[k] == 0:
                    del cnt[k]
        return res

File: problems/2610_convert_an_array_into_a_2d_array_with_conditions/Solution.java
import java.util.*;

class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        Map<Integer,Integer> cnt = new HashMap<>();
        for (int n : nums) cnt.merge(n, 1, Integer::sum);
        List<List<Integer>> res = new ArrayList<>();
        while (!cnt.isEmpty()) {
            List<Integer> row = new ArrayList<>(cnt.keySet());
            res.add(row);
            for (int k : row) { cnt.merge(k, -1, Integer::sum); if (cnt.get(k)==0) cnt.remove(k); }
        }
        return res;
    }
}
